services/profile/controller/controller.go

```diff
@@ -37,6 +42,24 @@ func GetProfile(w http.ResponseWriter, r *http.Request) {
 	json.NewEncoder(w).Encode(user_profile)
 }
 
+/*
+	GetProfileById is used to get a profile for a provided id.
+*/
+func GetProfileById(w http.ResponseWriter, r *http.Request) {
+	parameters := r.URL.Query()
+
+	id := parameters.Get("id")
```

Comment:
This should work, but for consistency with the other services, let's change this to

```id := mux.Vars(r)["id"]```